rclpy/rclpy/impl/implementation_singleton.py

```diff
@@ -27,6 +27,14 @@
 """
 
 import importlib
-
-
-rclpy_implementation = importlib.import_module('._rclpy', package='rclpy')
+import os
+
+try:
+    rclpy_implementation = importlib.import_module('._rclpy', package='rclpy')
+except ImportError as e:
+    if os.path.isfile(e.path):
+        raise ImportWarning(
```

Comment:
I don't think this should be a `ImportWarning`. I thought @sloretz was referring to call `warnings.warn(...)`.

Reply:
Yeah I figured I just liked the idea of the message being the last thing the user see rather than having to scroll up the stack trace to see our message burried in the middle.
But I agree it's less semantically correct

Reply:
You could also extend the `msg` of the raised exception `e` to add additional information if you prefer a single message.